cmd/mailchain/internal/http/handlers/messages.go

```diff
@@ -98,9 +99,13 @@ func GetMessages(inbox stores.State, receivers map[string]mailbox.Receiver, ks k
 					MessageID:   message.ID.HexString(),
 					ContentType: message.Headers.ContentType,
 				},
-				Read:    readStatus,
-				Subject: message.Headers.Subject,
-				Status:  "ok",
+				Read:                    readStatus,
+				Subject:                 message.Headers.Subject,
+				Status:                  "ok",
+				BlockId:                 transactionData.BlockId,
+				BlockIdEncoding:         encoding.TypeHex0XPrefix,
+				TransactionHash:         transactionData.Hash,
```

Comment:
Since `TransactionHash ` will be a string the `transactionData.Hash` will need to be encoded. You can use `encoding.EncodeZeroX`